File: src/commands/backup.js
// commands/backup.js — Data backup, restore, and export
import { rpc } from "@/lib/apiClient";

// CreateBackupDto: { output_path: string, database_url: string }
export const createBackup = (payload) =>
  rpc("create_backup", payload);

// RestoreBackupDto: { backup_path: string, database_url: string }
export const restoreFromBackup = (payload) =>
  rpc("restore_from_backup", payload);

// Returns: BackupFile[] sorted by created_at DESC
export const listBackups = (directory) =>
  rpc("list_backups", { directory });

// AutoBackupScheduleDto: { store_id, backup_directory, frequency: "daily"|"weekly",
//   time_of_day: "HH:MM", database_url, retain_last_n? }
export const scheduleAutoBackup = (payload) =>
  rpc("schedule_auto_backup", payload);

// Returns JSON array of inventory rows (for CSV download)
export const exportInventoryCsv = (storeId) =>
  rpc("export_inventory_csv", { store_id: storeId });

File: src-tauri/src/models/backup.rs
// ============================================================================
// BACKUP & EXPORT MODELS
// ============================================================================

use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize)]
pub struct BackupFile {
    pub filename:   String,
    pub path:       String,
    pub size_bytes: u64,
    pub created_at: String,
}

#[derive(Debug, Serialize)]
pub struct BackupResult {
    pub success:    bool,
    pub path:       String,
    pub size_bytes: u64,
    pub message:    String,
}

#[derive(Debug, Deserialize)]
pub struct CreateBackupDto {
    pub output_path:  String,
    pub database_url: String,
}

#[derive(Debug, Deserialize)]
pub struct RestoreBackupDto {
    pub backup_path:  String,
    pub database_url: String,
}

#[derive(Debug, Deserialize)]
pub struct AutoBackupScheduleDto {
    pub store_id:         i32,
    pub backup_directory: String,
    pub frequency:        String,
    pub time_of_day:      String,
    pub database_url:     String,
    pub retain_last_n:    Option<i32>,
}
